src/app/core/config/config.service.ts

```diff
@@ -18,16 +18,17 @@ export abstract class ConfigService {
   protected abstract halService: HALEndpointService;
 
   protected getConfig(request: RestRequest): Observable<ConfigData> {
-    const [successResponse, errorResponse] =  this.responseCache.get(request.href)
-      .map((entry: ResponseCacheEntry) => entry.response)
-      .partition((response: RestResponse) => response.isSuccessful);
-    return Observable.merge(
-      errorResponse.flatMap((response: ErrorResponse) =>
-        Observable.throw(new Error(`Couldn't retrieve the config`))),
-      successResponse
-        .filter((response: ConfigSuccessResponse) => isNotEmpty(response) && isNotEmpty(response.configDefinition))
-        .map((response: ConfigSuccessResponse) => new ConfigData(response.pageInfo, response.configDefinition))
-        .distinctUntilChanged());
+    const responses = this.responseCache.get(request.href).pipe(map((entry: ResponseCacheEntry) => entry.response));
+    const errorResponses = responses.pipe(
+      filter((response) => !response.isSuccessful),
+      mergeMap(() => observableThrowError(new Error(`Couldn't retrieve the config`)))
+    );
+    const successResponses = responses.pipe(
+      filter((response) => response.isSuccessful && isNotEmpty(response) && isNotEmpty((response as ConfigSuccessResponse).configDefinition)),
+      map((response: ConfigSuccessResponse) => new ConfigData(response.pageInfo, response.configDefinition))
```

Comment:
`distinctUntilChanged()` does not add anything here because it performs a simple `===` operation on the current and previous emitted value. These values are (newly created) ConfigData objects, which will never have the same in-memory address.